File: src/main/java/no/nav/sosialhjelp/soknad/business/soknadunderbehandling/BatchSoknadUnderArbeidRepository.java
package no.nav.sosialhjelp.soknad.business.soknadunderbehandling;


import no.nav.sosialhjelp.soknad.domain.SoknadUnderArbeid;

import java.util.List;

public interface BatchSoknadUnderArbeidRepository {

    List<Long> hentGamleSoknadUnderArbeidForBatch();
    void slettSoknad(Long soknadUnderArbeidId);
    List<SoknadUnderArbeid> hentForeldedeEttersendelser();

}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/repository/soknadunderarbeid/BatchSoknadUnderArbeidRepository.kt
package no.nav.sosialhjelp.soknad.repository.soknadunderarbeid

interface BatchSoknadUnderArbeidRepository {
    fun hentSoknadUnderArbeid(behandlingsId: String): SoknadUnderArbeid?
    fun hentSoknadUnderArbeid(soknadUnderArbeidId: Long): SoknadUnderArbeid?
    fun hentGamleSoknadUnderArbeidForBatch(): List<Long>
    fun slettSoknad(soknadUnderArbeidId: Long?)
    fun hentForeldedeEttersendelser(): List<SoknadUnderArbeid>
}
